### ISSUE TITLE
Formatări automate defectuoase

### ISSUE THREAD

**Issue by LadislauStrifler** [dexonline/dexonline#687]
Se mai întâmplă să existe unele situații neprevăzute în care sistemul forțează o anumită formatare, cum ar fi cazul recent observatei:
`@1.@ ...` → `@1....@`
Nu știu cum ar fi mai bine:

**1.** să punem caracterul de deblocare `„\”` în fața celor 3 puncte
**2.** să peticim mereu inadvertențele care apar

În cazul **2.** această listă ar trebui să fie permanent deschisă

**Comment by CatalinFrancu** [comment]
Avem și o discuție pe e-mail despre asta (vezi _Modificări la accente și apostrofuri_) din ianuarie. Acolo spuneam că într-adevăr `Str::migrateFormatChars()` mai rău strică. Încă nu am decis cum să reparăm asta:

1. Aplicăm activ stilul recomandat de Chicago Manual of Style, că punctuația moștenește formatarea frazei în ansamblu, nu a cuvântului anterior, sau
2. Nu ne mai băgăm deloc, dar atunci ne punem încrederea în OCR, care în mod sigur va greși des la puncte și virgule.

Nu am o opinie, doar să-mi ziceți.

**Comment by MateiGall** [comment]
Eu sunt pentru (1).

**Comment by LadislauStrifler** [comment]
@MateiGall dacă (1) însemnă propunerea lui @CatalinFrancu, atunci bineînțeles și eu consimt. :)

**Comment by CatalinFrancu** [comment]
M-am mai gândit la asta și văd cel puțin două cazuri în care sistemul ar greși dacă ar scoate punctuația în afara formatărilor:

1. Numerotarea sensurilor cu `@1.@`, `@2.@` etc. Cel puțin DEX și DLRLC au punctul cu bold.
2. Citate în DLRLC care se termină cu semnul întrebării/exclamării. Semnul ar trebui inclus în formatare (italic), fiind parte din citat.

Momentan voi implementa soluția (2) de mai sus, deci voi opri formatările automate existente. Dacă găsim o metodă bună să rezolvăm aceste cazuri particulare și altele, pot implementa atunci soluția (1).

**Comment by LadislauStrifler** [comment]
OK, dar sistemul nu poate fi învățat să nu greșească? Adică:
````C#
if (înainte.de.punct == digit) { formatareCuPunctCuTot(); }
````

**Comment by CatalinFrancu** [comment]
Ei, clar putem declara corect regulile: `/@[IVX]\\.@/` pentru numere romane etc. Mă întrebam doar cât de catastrofic va arăta codul.

**Comment by CatalinFrancu** [comment]
În prezent formatarea nu migrează în niciun fel. De clarificat ce dorim și dacă excepțiile pot fi tratate ușor.

**Comment by LadislauStrifler** [comment]
În primul rând, eu secvențele `$@... @$` le-aș schimba cu `@$...$@`, deși nu e nicio diferență, dar conceptual parcă ar merge mai bine **bold**-_italic_. :)

**Comment by CatalinFrancu** [comment]
Sigur, o punem pe lista de regexuri. :-)

**Comment by LadislauStrifler** [comment]
Un aspect interesant este faptul că la un moment dat în cadrul șirului se deschide o formatare de text, să zicem **@**(bold), după care (imediat, dar nu necesar) se deschide o alta _$_(italic). Acestea se pot închide în aceeași manieră, adică nu consecutiv. Eu văd aceasta o problemă de formatare, pentru că la corectarea MDA2 am dat peste o grămadă de inadvertențe `@~$ș'âță$@`,  `@~$xxx$xxx@`. Nu văd de ce ar trebui tilda să fie doar **bold** și nu _italic_. De asemenea orice alt șir de acest fel. 
Concret: formatările deschise trebuie să fie consecutiv aplicate și consecutiv închise. 
Ar mai fi și alte exemple: `#E:# $@gance^2@ + -(e)ală$`, ceea ce ar trebui să fie `#E:# @$gance^2$@ + $-(e)ală$`

**Comment by CatalinFrancu** [comment]
Da și nu... de exemplu în MDA2 definițiile încep așa:

`@$ab'il, ~ă$@ $#a#$`

Dar nu aș vedea o problemă dacă ar începe așa:

`$@ab'il, ~ă@ #a#$`

O abordare solidă ar fi să detaliem sintaxa fiecărui dicționar. Pentru MDA2 ar fi: cuvântul-titlu în bold italic, partea de vorbire în bold, etimologia și variantele în bold italic... Apoi sistemul poate închide și deschide formatări (la o adică le poate chiar rade și reface pe cele puse de editor). Dar asta înseamnă parser, e bătaie mare de cap.

**Comment by LadislauStrifler** [comment]
În exemplul dat de mine `#E:# $@gance^2@ + -(e)ală$`, **plusul** nu e italic în dicționar. :)
Da, cred că ar trebui un parser, neapărat! M-am ocupat de câteva zile de reformatarea definițiilor, pentru că cele introduse până în momentul afișării mesajului de atenționare la formatare aveau deficiențe grave. În momentul de față aproape că se poate face parsare. Mai sunt unele mici inadvertențe pe care mai vreau să le îndrept. De exemplu degeaba faci parser dacă definiția conține `#Pl:# -` în loc de `#Pl:# $~i$` sau chiar `#Pl:# $~ri$`... și multe alte exemple. :)
